src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java

```diff
@@ -378,7 +378,7 @@ public JsonToken nextToken() throws IOException
                 boolean returnEnd = _headContext.isStartHandled();
                 f = _headContext.getFilter();
                 if ((f != null) && (f != TokenFilter.INCLUDE_ALL)) {
-                    f.filterFinishArray();
+                    f.filterFinishObject();
```

Comment:
Yeah, at first, I thought about it just as a cosmetic change as `com.fasterxml.jackson.core.filter.TokenFilter` have the same implementations fo both array and objects. But, if we want do it properly here, we should probably have to copy some code and call proper methods in ID_END_OBJECT and ID_END_ARRAY.

It's definitely out of scope of this commit, so I can move it to another issue if you want.